src/unity/python/turicreate/toolkits/one_shot_object_detector/one_shot_object_detector.py

```diff
@@ -95,12 +95,12 @@ def _get_native_state(self):
 
         # We don't know how to serialize a Python class, hence we need to 
         # reduce the detector to the proxy object before saving it.
-        state['detector'] = state['detector'].__proxy__
+        state['detector'] = state['detector']._get_native_state()
         return state
 
     @classmethod
     def _load_version(cls, state, version):
-        assert(version == _PYTHON_ONE_SHOT_OBJECT_DETECTOR_VERSION)
+        assert(version == cls._PYTHON_ONE_SHOT_OBJECT_DETECTOR_VERSION)
         # we need to undo what we did at save and turn the proxy object
         # back into a Python class
         state['detector'] = _ObjectDetector(state['detector'])
```

Comment:
Have you tested this? Doesn't this need to decode the native state back into an OD model first?

Reply:
Yeah, I tested `get_augmented_images` and checked for the `detector` key in it and it seemed to work!

Reply:
It doesn't suffice for the `detector` key to exist. The value must be a (functioning) object detector.

Reply:
Fixed!